src/documentation/sdks/euphoria.md

```diff
@@ -18,72 +18,530 @@ See the License for the specific language governing permissions and
+BeamFlow flow = BeamFlow.of(pipeline);
 
 // Source of data loaded from Beam IO.
 PCollection<String> input =
-    pipeline.apply(Create.of(inputs)).setTypeDescriptor(TypeDescriptor.of(String.class));
+  pipeline
+    .apply(Create.of(textLineByLine))
+    .setTypeDescriptor(TypeDescriptor.of(String.class));
 // Transform PCollection to euphoria's Dataset.
 Dataset<String> lines = flow.wrapped(input);
 
 // FlatMap processes one input element at a time and allows user code to emit
 // zero, one, or more output elements. From input lines we will get data set of words.
-Dataset<String> words = FlatMap.named("TOKENIZER")
+Dataset<String> words =
+  FlatMap.named("TOKENIZER")
     .of(lines)
-    .using((String line, Collector<String> context) -> {
-      for (String word : line.split("\\s+")) {
+    .using(
+      (String line, Collector<String> context) -> {
+        for (String word : Splitter.onPattern("\\s+").split(line)) {
         context.collect(word);
       }
-    })
+      })
     .output();
 
-// From each input element we extract a key (word) and value, which is the constant `1`.
-// Then, we reduce by the key - the operator ensures that all values for the same
-// key end up being processed together. It applies user defined function (summing word counts for each
-// unique word) and its emitted to output. 
-Dataset<Pair<String, Long>> counted = ReduceByKey.named("COUNT")
+// Now we can count input words - the operator ensures that all values for the same
+// key (word in this case) end up being processed together. Then it counts number of appearances
+// of the same key in 'words' dataset and emits it to output.
+Dataset<KV<String, Long>> counted =
+  CountByKey.named("COUNT")
     .of(words)
     .keyBy(w -> w)
-    .valueBy(w -> 1L)
-    .combineBy(Sums.ofLongs())
     .output();
 
 // Format output.
-Dataset<String> output = MapElements.named("FORMAT")
+Dataset<String> output =
+  MapElements.named("FORMAT")
     .of(counted)
-    .using(p -> p.getFirst() + ": " + p.getSecond())
+    .using(p -> p.getKey() + ": " + p.getValue())
     .output();
 
 // Transform Dataset back to PCollection. It can be done in any step of this flow.
 PCollection<String> outputCollection = flow.unwrapped(output);
 
 // Now we can again use Beam transformation. In this case we save words and their count
 // into the text file.
-outputCollection.apply(TextIO.write().to(options.getOutput()));
+outputCollection.apply(TextIO.write().to("counted_words"));
 
 pipeline.run();
 ```
 
+## Euphoria Guide
+### `Flow` your data
+Euphoria API is composed as a flow of operators called `BeamFlow`. There is a set of operators which allows to construct `BeamFlow` according to your Big Data application needs. `BeamFlows` are easily created from Beam `Pipeline`.
+```java
+Pipeline pipeline = Pipeline.create(options);
+
+BeamFlow flow = BeamFlow.of(pipeline);
+```
+Or from `PCollection`. That is especially useful when some of many Beam's IO are utilized as data inputs.
```

Comment:
Or from `PCollection`, which is especially useful when any of Beam's IO are utilized as data inputs.

Reply:
Thanks.